blackbox/docs/administration/ssl.txt

```diff
@@ -114,3 +114,272 @@ string. E.g.::
+Command::
+
+    keytool -keystore keystore -genkey -alias server -validity 36500
+
+Output::
+
+    Enter keystore password:
+    Re-enter new password:
+    What is your first and last name?
+      [Unknown]:  CrateDB
+    What is the name of your organizational unit?
+      [Unknown]:  PSQL Netty SSL
+    What is the name of your organization?
+      [Unknown]:  crate.io
+    What is the name of your City or Locality?
+      [Unknown]:  Dornbirn
+    What is the name of your State or Province?
+      [Unknown]:  Vorarlberg
+    What is the two-letter country code for this unit?
+      [Unknown]:  AT
+    Is CN=CrateDB, OU=PSQL Netty SSL, O=crate.io, L=Dornbirn, ST=Vorarlberg, C=AT correct?
+      [no]:  yes
+
+    Enter key password for <server>
+        (RETURN if same as keystore password):
+    Re-enter new password:
+
+
+Generate a Certificate Signing Request
+--------------------------------------
+
+To establish trust for this key, we need to sign it. This is done by generating
+a certificate signing request. If you have access to a certificate authority (CA),
+you can skip the next optional steps and get the signed certificate from it using
+the signing request which we will generate in the following. If you don't, then
+follow the optional steps after this step to establish your own CA.
+
+Command::
+
+    keytool -keystore keystore -certreq -alias server -keyalg rsa -file server.csr
+
+
+Output::
+
+    Enter keystore password:
+    Enter key password for <server>
+
+
+Optional: Generate a Self-Signed Certificate to act as a Certificate Authority (CA)
+-----------------------------------------------------------------------------------
+
+If you don't get your certificate signed from one of the official CAs,
+you might want to create your own CA with a self-signed certificate:
+
+Command::
+
+    openssl req -x509 -sha256 -nodes -days 36500 -newkey rsa:2048 -keyout rootCA.key -out rootCA.crt
+
+
+Output::
+
+    Generating a 2048 bit RSA private key
+    .......................................................................+++
+    .............................................................+++
+    writing new private key to 'rootCA.key'
+    -----
+    You are about to be asked to enter information that will be incorporated
+    into your certificate request.
+    What you are about to enter is what is called a Distinguished Name or a DN.
+    There are quite a few fields but you can leave some blank
+    For some fields there will be a default value,
+    If you enter '.', the field will be left blank.
+    -----
+    Country Name (2 letter code) [AU]:AT
+    State or Province Name (full name) [Some-State]:Vorarlberg
+    Locality Name (eg, city) []:Dornbirn
+    Organization Name (eg, company) [Internet Widgits Pty Ltd]:Crate.io
+    Organizational Unit Name (eg, section) []:Cryptography Department
+    Common Name (e.g. server FQDN or YOUR name) []:crate.io
+    Email Address []:info@crate.io
+
+
+Optional: Generate Signed Cert
```

Comment:
IMHO this is a required step. otherwise the keystore/truststore won't contain any certificate to share with.